server/src/main/java/org/opensearch/indices/IntKeyLookupStore.java

```diff
@@ -0,0 +1,164 @@
+     * @param value The value to check.
+     * @return true if the value was found, false otherwise. Due to collisions, false positives are
+     * possible, but there should be no false negatives unless forceRemove() is called.
+     */
+    boolean contains(int value) throws Exception;
+
+    /**
+     * Returns the transformed version of the input value, that would be used to stored it in the keystore.
+     * This transformation should be always be the same for a given instance.
+     * @param value The value to transform.
+     * @return The transformed value.
+     */
+    int getInternalRepresentation(int value);
+
+    /**
+     * Attempts to safely remove a value from the internal structure, maintaining the property that contains(value)
+     * will never return a false negative. If removing would lead to a false negative, the value won't be removed.
+     * Classes may not implement safe removal.
+     * @param value The value to attempt to remove.
+     * @return true if the value was removed, false if it wasn't.
+     */
+    boolean remove(int value) throws Exception;
+
+    /**
+     * Check if the implementing class supports safe removals. If it doesn't, remove() will always return false.
+     * @return true if the class supports safe removals, false if it doesn't.
+     */
+    boolean supportsRemoval();
+
+    /**
+     * Remove the transformed version of this value from the store. Calling this function may cause
+     * contains() to return false negatives for future values.
+     * @param value The value to forcibly remove.
+     */
+    void forceRemove(int value) throws Exception;
+
+    /**
+     * Check if the object currently guarantees having no false negatives when running contains().
+     * @return false if there will not be false negatives, true if there could be false negatives.
+     */
+    boolean canHaveFalseNegatives();
+
+    /**
+     * Returns the number of distinct values stored in the internal data structure.
+     * Does not count values which weren't successfully added due to collisions.
+     * @return The number of values
+     */
+    int getSize();
+
+    /**
+     * Returns the number of times add() has been run, including unsuccessful attempts.
+     * @return The number of adding attempts.
+     */
+    int getNumAddAttempts();
+
+    /**
+     * Returns the number of times add() has returned false due to a collision.
+     * @return The number of collisions.
+     */
+    int getNumCollisions();
+
+    /**
+     * Returns the current internal data structure.
+     * @return A string representing the currently used internal data structure.
+     */
+    String getCurrentStructure() throws Exception;
+
+    /**
+     * If the structure uses a roaring bitmap with a modulo at some point, returns that modulo.
+     * If it uses a RBM without a modulo or doesn't use an RBM, returns 0.
+     * @return The modulo.
+     */
+    int getModulo();
```

Comment:
Hmm don't think we need to expose this. If we use modulo 2^22 for example, the concrete class can itself can be called RoaringBitMapModule22 or something like that. I don't see a use case for this function.